pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -655,6 +649,36 @@ func (r *ReconcileSyncSetInstance) applySyncSetResources(ssi *hivev1.SyncSetInst
 	return nil
 }
 
+func applyResource(resource runtime.RawExtension, logger log.FieldLogger, h Applier) error {
```

Comment:
I extracted this function as I did not like the way the error handling was looking otherwise, we went from one possible error to three, all of which need to result in some common handling to store the result in the condition.